### PR TITLE
Muscle

### PR DESCRIPTION
**Pull request by MGordon09**
- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: tests/software/muscle/main.nf
Comment (KevinMenden) [inline tests/software/muscle/main.nf:10]: It would be nice to have a file on the modules test data repository for testing, of course.
You just need a mult-fasta file, right? I just had a look and we have a multi fasta file here:
https://github.com/nf-core/test-datasets/blob/modules/data/genomics/sarscov2/illumina/fasta/contigs.fasta

It's contigs so not really suitable for an MSA of course, but as long as MUSCLE doesn't crash it should be fine.
Could you have a look at that? Otherwise we could add test data for this. Should be fairly straight forward.

**Review by KevinMenden — COMMENTED** [review]
Thanks @MGordon09 , this looks really good!
The only comment I have is about the test data.

File: tests/software/muscle/main.nf
Comment (MGordon09) [reply tests/software/muscle/main.nf:10]: Hi @KevinMenden! 

Thanks a lot for taking a look. I first tried the fasta files available on the modules test data repo, but unfortunately MUSCLE was crashing so I selected something from another test-data repo. It's no problem at all for me to create this test data; I'll let you know when it is ready

Thanks again for the help!

Comment (KevinMenden) [reply tests/software/muscle/main.nf:10]: Amazing, thanks!

Comment (MGordon09) [reply tests/software/muscle/main.nf]: Hi @KevinMenden 

Had another look at the modules test repo and found something suitable (I somehow missed `informative_sites_fas` first time round!!!). Hope this is ok and does the trick.

**Comment by MGordon09** [comment]
Hi @KevinMenden ,

Thank you so much for reviewing the `metaphlan3` module and all your helpful suggestions. I just wanted to let you know I have swapped out MUSCLE test data for something available on the modules test repo, so hopefully this should be good to go.

Apologies for the late email on a Friday evening, I am on a break next week so will not be around  :)

Thank you again and enjoy your weekend!

File: software/muscle/main.nf
Comment (grst) [inline software/muscle/main.nf:26]: As I understand from your documentation, these two belong together and only one of them is emitted at a time? 
In that case consider merging the output channel `"*.{phyi,phys}"`. 
Otherwise, please document every output channel separately.

File: software/muscle/main.nf
Comment (grst) [inline software/muscle/main.nf:21]: I think general consensus is to pass a map `val(meta)` through every module, have a look at e.g. `fastqc` how this is done.

File: software/muscle/meta.yml
Comment (grst) [inline software/muscle/meta.yml:21]: I think the id of each channel should match the `emit`. 
But I'm not sure how strictly we have been checking this in the past. 

@KevinMenden, can/should we lint for that?

Comment (MGordon09) [reply software/muscle/meta.yml]: Have implemented this in meta.yml. Thanks for pointing it out - definitely improves documentation clarity!

File: software/muscle/main.nf
Comment (MGordon09) [reply software/muscle/main.nf]: Apologies Gregor, my documentation was misleading.  MUSLCE can produce either or both phyi and phys output file in a single run - have updated meta.yml to reflect this

File: software/muscle/main.nf
Comment (MGordon09) [reply software/muscle/main.nf]: Thanks for pointing out - have updated

**Comment by MGordon09** [comment]
Hi @grst 

Thank you for taking some time to look at the module and your excellent feedback and suggestions. I have implemented these changes, please let me know if there is anything else required.

Thanks again, 
Martin

File: software/muscle/meta.yml
Comment (KevinMenden) [reply software/muscle/meta.yml]: @grst no we don't, but it would probably be good, have seen something similar lately as well :+1:

**Comment by grst** [comment]
Cool, @MGordon09! 
This LGTM now, but let's maybe wait for another pair of eyes for final approval.

**Review by KevinMenden — APPROVED** [review]
Thanks! LGTM as well!

**Pull request merged**
